test/end2end_test.go

```diff
@@ -1334,28 +1334,30 @@ func testConcurrentServerStopAndGoAway(t *testing.T, e env) {
 }
 
 func (s) TestDetailedConnectionCloseErrorPropagatesToRpcError(t *testing.T) {
+	rpcStartedOnServer := make(chan struct{})
 	rpcDoneOnClient := make(chan struct{})
 	ss := &stubserver.StubServer{
 		FullDuplexCallF: func(stream testpb.TestService_FullDuplexCallServer) error {
+			close(rpcStartedOnServer)
```

Comment:
FWIW, another option here would be to send a message, and have the client receive it.

Reply:
This behavior of this test is IMO easier to reason about as is, just since there's slightly fewer moving parts and less happening on the wire. So for the purposes of this test, I feel the existing approach is simpler so I'll keep as is.